### PR TITLE
BackendSrv: Improves logic for hiding requests from query inspector

### PR DESCRIPTION
**Pull request by torkelo**
After the backendSrv refactoring where I unified the requests paths for all http api calls it created this issue where meta analytics events would show up after a few seconds in the QueryInspector. 

This PR 
* Changes name of the silent bool to hideFromInspector to be more descriptive 
* When unset the auto logic will set this to true for all local non-data query calls

### CODE REVIEW

**Review by ivanahuckova — COMMENTED** [review]
LGTM! Just one thing, we have added Query Inspector to Explore and the naming `silent -> hideFromInspector` should be changed [in there](https://github.com/grafana/grafana/blob/master/public/app/features/explore/ExploreQueryInspector.tsx#L18) as well.

**Comment by torkelo** [comment]
I do not understand why that query inspector is different, it should use exactly the same component

**Comment by torkelo** [comment]
Created a followup issue to fix the explore query inspector https://github.com/grafana/grafana/issues/26893 

I gave feedback on the explore inspect issue that the query inspectors needed to be unified, do not want 90% duplicated components for that.

**Comment by ivanahuckova** [comment]
👍 yeah, makes sense. But, for the current state, I think that it makes sense to update naming `silent -> hideFromInspector` so it works on current master in both cases, before we unify the inspectors.

**Review by ivanahuckova — APPROVED** [review]
LGTM🕺

File: public/app/core/specs/backend_srv.test.ts [hunk 1]
```diff
@@ -83,21 +83,22 @@ describe('backendSrv', () => {
...
-      ${1}         | ${'/api/dashboard/'}                     | ${{ Authorization: 'Some Auth' }} | ${1}         | ${undefined}   | ${{ retry: 1, url: 'api/dashboard/', headers: { 'X-DS-Authorization': 'Some Auth', 'X-Grafana-Org-Id': 1 } }}
-      ${1}         | ${'/api/dashboard/'}                     | ${{ Authorization: 'Some Auth' }} | ${1}         | ${true}        | ${{ retry: 1, url: 'api/dashboard/', headers: { 'X-DS-Authorization': 'Some Auth', 'X-Grafana-Org-Id': 1, 'X-Grafana-NoCache': 'true' } }}
+      ${undefined} | ${'http://localhost:3000/api/dashboard'} | ${undefined}                      | ${undefined} | ${undefined}   | ${{ hideFromInspector: false, retry: 0, url: 'http://localhost:3000/api/dashboard' }}
+      ${1}         | ${'http://localhost:3000/api/dashboard'} | ${{ Authorization: 'Some Auth' }} | ${1}         | ${true}        | ${{ hideFromInspector: false, retry: 1, url: 'http://localhost:3000/api/dashboard', headers: { Authorization: 'Some Auth' } }}
```
Comment (dprokop) [inline public/app/core/specs/backend_srv.test.ts:87]: @torkelo shouldn't these two be hidden?

Comment (torkelo) [reply public/app/core/specs/backend_srv.test.ts:87]: no, absolute urls are assumed to be external data source urls (when data source access is set to Proxy/browser)

Comment (dprokop) [reply public/app/core/specs/backend_srv.test.ts:87]: OK, this makes sense then, thank you

**Review by dprokop — APPROVED** [review]

**Pull request merged**